### DISCUSSION TITLE
pyEXP orbit integration reference frame

### DISCUSSION THREAD

**Discussion by jngaravitoc** [EXP-code/EXP#102] [General]
When computing orbits of tracers in pyEXP, given a set of coefficients and the center of the expansion, is there a way to specify the reference frame of the integration? For example, I would like to experiment with integrating orbits in a fixed reference frame and in a moving frame.  
Moving the discussion with @chervs @M1ssing-N0 @The9Cat to this space.

**Comment by The9Cat** [comment]
Thanks for moving the discussion here @jngaravitoc 

The current design idea is:
- The user writes an acceleration functor that gets passed to `IntegrateOrbits`
- A Python function or lambda closure supplied to `IntegrateOrbits` should have full access to your variable scope so you should be able to do all sorts of 'what ifs'
- This function can use the `FieldGenerator` evaluation for the desired expansion fields in whatever frame is desired by applying boosts and transformations

My thinking here was:
- Can not possibly accommodate the large number of possibilities that a user might need
- So better to provide the building blocks with examples
- Consider the `AllTimeAccel` method to be a placeholder for a custom force.  See the `orbits with a custom force` notebook in `pyEXP-examples/Tutorials/Orbits` for an example of this approach

As people use this, we may want to provide more hardwired features or provide additional interaction methods, not sure.  The current implementation was meant to be flexible.  Feel free to make suggestions if there's something you need to do but can't.

**Reply by jngaravitoc** [reply]
I think this new acceleration function is what we need for the needs that we were discussing with @chervs  and @M1ssing-N0. 
Thanks @The9Cat!

**Comment by The9Cat** [comment]
Just to be clear, this is not a **new** feature; it's in the current release and has been there for a beginning.  So you can use it now.
